vendor/k8s.io/kube-openapi/pkg/handler/handler.go

```diff
@@ -55,13 +55,38 @@ type OpenAPIService struct {
 
 	lastModified time.Time
 
-	specBytes []byte
-	specPb    []byte
-	specPbGz  []byte
+	openapiSpec *spec.Swagger
+	specBytes   []byte
+	specPb      []byte
+	specPbGz    []byte
 
 	specBytesETag string
 	specPbETag    string
 	specPbGzETag  string
+
+	jcache    jsonCache
```

Comment:
All these cache shouldn't really have to be different "types", they can all be just `Cache`.